bugRzilla_review.Rmd

```diff
@@ -168,45 +215,659 @@ Also there are columns which are empty so they are of no use of the analysis lik
+colnames(created_month_df) <- c("creation_month", "Bug_count")
+
+# last modified bugs per month
+lastdiffed_month_df <- data.frame(total_data$bug_id, 
+                                  total_data$lastdiffed_month)
+lastdiffed_month_df <- aggregate(total_data.bug_id~., 
+                                 lastdiffed_month_df, 
+                                 function(x) length(unique(x)))
+colnames(lastdiffed_month_df) <- c("lastdiffed_month", "Bug_count")
+
+# created and last modified bugs per month in single dataframe
+lastdiffed_month_df <- pad(lastdiffed_month_df)
+cre_last_month <- data.frame(created_month_df, 
+                             lastdiffed_month_df)
+skim(cre_last_month)
+
+# last modified bugs per week
+created_week_df <- data.frame(total_data$bug_id, 
+                              total_data$creation_week)
+created_week_df <- aggregate(total_data.bug_id~., 
+                             created_week_df, 
+                             function(x) length(unique(x)))
+colnames(created_week_df) <- c("creation_week", "Bug_count")
+
+# last modified bugs per week
+lastdiffed_week_df <- data.frame(total_data$bug_id, 
+                                 total_data$lastdiffed_week)
+lastdiffed_week_df <- aggregate(total_data.bug_id~., 
+                                lastdiffed_week_df, 
+                                function(x) length(unique(x)))
+colnames(lastdiffed_week_df) <- c("lastdiffed_week", "Bug_count")
+
+# created and last modified bugs per week in single dataframe
+created_week_df <- pad(created_week_df)
+lastdiffed_week_df <- pad(lastdiffed_week_df)
+cre_last_week <- data.frame(created_week_df, 
+                            lastdiffed_week_df)
+skim(cre_last_week)
+```
+
+### Visualizations
+
+```{r cre_last_year_graph}
+cre_last_year_graph <- cre_last_year %>%
+    ggplot() +
+    geom_line(aes(x = creation_year, 
+                  y = Bug_count,
+                  colour = "creation_year")) +
+    geom_line(aes(x = lastdiffed_year,
+                  y = Bug_count.1, 
+                  color = "lastdiffed_year")) +
+    labs(
+        title = "Year in which bugs are Created vs Last modified",
+        x = "year",
+        y = "Bug_Count"
+    )
+
+ggplotly(cre_last_year_graph)
+```
+<font size = 4>
+    The Time-series graph is the relationship between the `bug_count` and the `year` in which the bug is `created` and `last_modified`. The aim is to check in every year how many bugs are `created` and `last modified`. In the year `2015`, a total of `470` bugs are created and `2015`, `818` bugs are modified which is a sudden peak from the year `2014`. From the year `2007` to `2013` the is a gradual peak but from year `2015` there is a sudden downfall in `2017` from `470` to `172` bugs are created and from `2015` to `2016` there is a sudden downfall from `818` to `287` and again `2018` to `2019` there is an increase in the modification of the bugs from `240` to `595`.
```

Comment:
I would not read too much into minor fluctuations from month to month. It is more interesting that up to 2010, the number of bugs created is similar to the number of bugs modified, suggesting that R core were keeping pace with bug reports. From 2010 to 2016 the number created increased relative to the number modified, suggesting a backlog was being created. The big spikes that you mention are suspicious: was there some housekeeping operation done to many bugs? These spikes will drive the pattern seen in the yearly plot above. If this spikes are spurious, i.e. an artefact of a housekeeping operation rather than activity reviewing/fixing bugs, then last_modified may not be the best variable to use here. Bugs closed would probably be more relevant.

Reply:
Nice observation, but with the provided data one cannot be certain whether the `housekeeping` was done in the perceivable period. In the data, only the following columns were present viz. 

- bug_when
- fieldid
- added
- removed
- modification_time
- description
- mimetype
- ispatch
- filename
- submitter_id
- isobsolete
- isprivate
- assigned_to
- product_id
- reporter
- creation_year
- creation_month
- creation_week
- lastdiffed_year
- lastdiffed_month
- lastdiffed_week

Intuitively I selected the lastdiffed_* columns to represent the patching date.

Reply:
Well first of all, can you look at some examples at those peak times, to see if my suspicion is correct about it being some housekeeping operation? Maybe some variable is changed in bulk - if that's the case, it would good to add this information to the report, to explain the peaks seen in the plot. Even if you can't track down the reason, the possibility that these peaks are due to some housekeeping operation should be discussed.

If you can track down the cause, it might be possible to define a filter for these changes, so that they can be ignored in the plot. However, when you merge all the tables into `total_data`, you also have the `bug_status`. So it should be possible to restrict the count of `lastdiffed_month` to cases when `bug_status` is "CLOSED" or perhaps to the earliest `lastdiffed_month` when `bug_status` is "CLOSED", to get the month when the bug was closed.

Reply:
Thanks for pointing that out it might be helpful to restrict the count of `lastdiffed_month` to cases when `bug_status` is "CLOSED".

Reply:
I tried but to restrict the count of `lastdiffed_month` to cases when `bug_status` is "CLOSED" from that we can't conclude why the big spikes were there.

Reply:
But knowing when the big spikes is already useful. Probably these are migrations between bug trackers or when an R-core member is actively managing (old?) issues.